### ISSUE TITLE
No way for update arguments via IMAGES_HOOK anymore

### ISSUE THREAD

**Issue by kvaps** [OpenNebula/one#2626]
**Description**
Since `5.6` we have nice locking mechanism. But all new images created with LOCKING enabled.
So if you have some hook which adds some custom parameters to creating images, it will not work anymore.

`IMAGE_HOOK` supports only `CREATE` and `DELETE` operations, no way to call it after image is `READY`

**To Reproduce**

* Write hook for adding specific label to the image:
```
#!/bin/sh
EDITOR="echo LABELS=mylabel >" oneimage update --append $1
```
* Install hook in `oned.conf`:
```
IMAGE_HOOK = [
    name      = "add_label",
    on        = "CREATE",
    command   = "add_label.sh",
    arguments = "$ID" ]
```
* Upload new image, alpine for example:
```
http://dl-cdn.alpinelinux.org/alpine/v3.8/releases/x86_64/alpine-standard-3.8.1-x86_64.iso
```
* In logs you will see error
```
[Z0][ImM][I]: Copying http://dl-cdn.alpinelinux.org/alpine/v3.8/releases/x86_64/alpine-standard-3.8.1-x86_64.iso to repository for image 1440
[Z0][ReM][E]: Req:6704 UID:0 one.image.update result FAILURE [one.image.update] User [0] : IMAGE is locked. [1440].
```

**Expected behavior**
Attached label to the image

**Details**
 - Affected Component: Core
 - Hypervisor: KVM
 - Version: 5.6.0

## Progress Status
- [ ] Branch created 
- [ ] Code committed to development branch
- [ ] Testing - QA
- [ ] Documentation 
- [ ] Release notes - resolved issues, compatibility, known issues
- [ ] Code committed to upstream release/hotfix branches
- [ ] Documentation committed to upstream release/hotfix branches

**Comment by atodorov-storpool** [comment]
Same problem when importing an image issue #2292

**Comment by kvaps** [comment]
For now I use in my hook as workaround, it is working fine
```
oneimage unlock "$1"
```

**Comment by atodorov-storpool** [comment]
Thanks @kvaps !
Works in _datastore/mad/cp_ too :+1: 
I've added re-lock after the update though. Just for completeness :)

**Comment by kvaps** [comment]
>re-lock after the update though
I think it's not necessary, the image have LOCKED state itself, but you can have hypothetic situation, when image already uploaded and unlocked by opennebula, then your hook finishes and locking this image again.
That's why I've decided to not lock image afterwards.

**Comment by juanmont** [comment]
Another possibility you have is to make a synchronization in your hook, waiting for the image to be ready.

We did this Feature because there were enough people who needed the image to be blocked until it was ready.

We are going to close this issue. Thanks

**Comment by kvaps** [comment]
@juanmont what's the problem? images still can't be used even if it no locked via `one.image.lock`, it still have `LOCKED` state, and can't be used by user.

> make a synchronization in your hook, waiting for the image to be ready.

That's crazy, this functional shouldn't be implemented on the hook side.
We need normal hooking system, which allows to call the hook after image is `READY`.

**Comment by juanmont** [comment]
Hello @kvaps ,

We have the following problem with this topic, OpenNebula has 3 kinds of lock  ( Use, Manage and Admin ). We need to lock an image until is READY, because, we don't want that this image can be used until OpenNebula has finished with it.

The problem is that we can't lock an image only for USE and allow MANAGE over this image, so USE_LOCK lock also lock manage actions and admin action.

You can unlock the image into you hook and re-lock this image when you finished your changes.

**Comment by rsmontero** [comment]
solved via #2770.

Thanks!
